File: client/src/game/conf/BgImgConfig.ts
class BgImgConfig {
    private static _dic:Dictionary<BgImgConfig> = new Dictionary<BgImgConfig>();
    private static _arr:Array<BgImgConfig> = new Array<BgImgConfig>();

	 public Id :number ;
     public Name :string ;
     public Location : Array<number>  =new Array<number>();
    
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<BgImgConfig>{
        return  BgImgConfig._dic;
    }
    public static get Arr():Array<BgImgConfig>{
        return BgImgConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:BgImgConfig = <BgImgConfig>json[i];
				BgImgConfig._arr.push(config);
				BgImgConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("BgImgConfig loader fail!");
        }
    }
}

File: server/src/game/conf/BgImgConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type BgImgConfigElem struct {
	Id int
	Name string
	Location []int
	Des string
	
}

type BgImgConfig0 struct {
	Dic map[int]*BgImgConfigElem
	Arr []*BgImgConfigElem
}

var (
	BgImgConfig BgImgConfig0
)

func loadBgImgConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	BgImgConfig.Dic = make(map[int]*BgImgConfigElem)
	BgImgConfig.Arr = make([]*BgImgConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &BgImgConfig.Arr)
	for _,conf := range BgImgConfig.Arr{
		BgImgConfig.Dic[conf.Id] = conf
	}
	
}
